bluemira/neutronics/constants.py

```diff
@@ -0,0 +1,84 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+"""constants used for the neutronics module"""
+
+from periodictable import elements
+
+from bluemira.base.constants import N_AVOGADRO, raw_uc
+
+# Amount of energy released in a single dt fusion reaction, in MeV.
+energy_per_dt = raw_uc(17.58, "MeV", "J")
+# Amount of energy carried away by the neutron
+dt_neutron_energy = energy_per_dt * (4 / 5)  # [J]
```

Comment:
No problem, not a fan of hard-coded magic numbers with no explanation of origin (I only used this because it was already present in the legacy code) so happy to make this change.

However I would like to point out that, accuracy-wise, it's not going to make a lot of difference. In fact this code makes quite a number of assumptions that causes a lot more inaccuracies, starting with:
1. 100% of reactions are DT reactions; and the average energy released by them is the same as the energy released by DT collisions happening in CoM frame.
2. 100% of the neutron energy will be converted to heat; and 0% of the alpha's energy will be converted to heat
So there are much larger error sources out there to begin with. (not an excuse to not prune away these smaller error sources though, hence I did make the change).

Reply:
Solved in fb74c2d844fd628430bc1fbd461d902c60402a60.

Note to self: Shall talk to Tim Eade/James Hague about Assumption 2 above and then remove the `calculate_total_neutron_rate` method from params.py and neutronics_axisymmetric.py; and replace with `plasma_physics.reactions.n_DT_reactions`

Reply:
Yeah definitely makes almost no difference, I just think it is nice for people to stumble across the source code and see the "proper" calculation.

Reply:
Slight update as I've made the edit:
I previously thought Assumption 2 was
"100% of the neutron energy will be converted to heat; and 0% of the alpha's energy will be converted to heat"
But actually after looking at the code more carefully, Assumption 2 was actually
"100% of the neutron energy will be converted to heat; and also 100% of the alpha's energy will be converted to heat".
Seeing this, I've made further simplifications to the code by replacing "calculate_total_neutron_rate" with "n_DT_reactions" in a18ec2bc5abc5fb264b67e82b7c718c803b14f66.